vendor/bat-native-ledger/src/bat/ledger/internal/database/database_server_publisher_info.cc

```diff
@@ -105,77 +138,53 @@ bool DatabaseServerPublisherInfo::MigrateToV15(
   return banner_->Migrate(transaction, 15);
 }
 
-void DatabaseServerPublisherInfo::DeleteAll(ledger::ResultCallback callback) {
-  auto transaction = ledger::DBTransaction::New();
-  const std::string query = base::StringPrintf("DELETE FROM %s", kTableName);
-
-  auto command = ledger::DBCommand::New();
-  command->type = ledger::DBCommand::Type::EXECUTE;
-  command->command = query;
-  transaction->commands.push_back(std::move(command));
-
-  auto transaction_callback = std::bind(&OnResultCallback,
-      _1,
-      callback);
-
-  ledger_->RunDBTransaction(std::move(transaction), transaction_callback);
-}
+bool DatabaseServerPublisherInfo::MigrateToV23(
+    ledger::DBTransaction* transaction) {
+  DCHECK(transaction);
 
-void DatabaseServerPublisherInfo::InsertOrUpdatePartialList(
-    const std::vector<ledger::ServerPublisherPartial>& list,
-    ledger::ResultCallback callback) {
-  if (list.empty()) {
-    callback(ledger::Result::LEDGER_OK);
-    return;
+  // TODO(zenparsing): Do we need to vacuum after this? Vacuum
+  // took 276ms and reduced the size from 137MB to 16MB.
```

Comment:
@NejcZdovc After emptying the data from `server_publisher_info`, I think we need to vacuum to release disk space. Do we currently do any vacuuming?

Reply:
Would be great to add vacuum type in DB transaction so we can use it different situations

Reply:
I added a `VACUUM` DB command type. I decided to include it as part of this PR since only this PR exercises it.